TextEditor/SelectPositions.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Security.AccessControl;
+
+namespace TextEditor
+{
+    //the class responsible for the selected positions in the text 
+    public static class SelectPositions
```

Comment:
The responsibility of this class is to handle location for one text, right? that means that one instance of `SelectPositions` class must be bound to one instance of `ClassTextEditor`. while `SelectPositions` is static - absolutely any code can change position related to one specific text - and the editor working with this text will work incorrectly.

1. please make this class not static, and don't add static members to it.
2. in general, static classes, methods, or properties should be your last choice - so better not use them at all and ask me when you can't make it work without static.

Reply:
I made this class not static...I will push a little later